doc/esmvalcore/fixing_data.rst

```diff
@@ -0,0 +1,232 @@
+
+They are automatically loaded and applied when the dataset is preprocessed.
+
+Fixing a dataset
+================
+
+To illustrate the process of creating a fix we are going to construct a new
+one from scratch for a fictional dataset. We need to fix a CMIP7 model
+called PERFECT-MODEL that is reporting a missing latitude coordinate for
+variable tas.
+
+Check the output
+----------------
+
+Next to the error message, you should see some info about the iris cube: size,
+coordinates. In our example it looks like this:
+
+.. code-block:: python
+
+   cube output with latitude called altitude
+
+
+So now the mistake is clear: the latitude coordinate is badly named and the
+fix should just rename it
+
+Create the fix
+--------------
+
+We start by creating the module file. In our example the path will be
+``esmvaltool/cmor/_fixes/CMIP7/PERFECT_MODEL.py``. If it already exists
+just add the class to the file, there is no limit in the number of fixes
+we can have in any given file
+
+Then we have to create the class for the fix deriving from
+:class:`esmvalcore.cmor._fixes.Fix`
+
+.. code-block:: python
+
+    """Fixes for PERFECT-MODEL."""
+    from esmvalcore.cmor.fix import Fix
+
+    class tas(Fix):
+         """Fixes for tas variable.""""
+
+And now we must choose the method to use between the ones offered by the
+Fix class:
+
+- ``fix_file`` : should be used only to fix errors that prevent data loading.
+  As a rule of thumb, you should only use it you are not even
+  reaching the checks
+
+- ``fix_metadata`` : you want to change something in the cube that is not
+  the data.
+
+- ``fix_data``: you need to fix the data. Beware: coordinates data values are
+  part of the metadata
+
+In our case we need to rename the coordinate ``altitude`` to ``latitude``,
+so we will implement the ``fix_metadata`` method
+
+.. code-block:: python
+
+    """Fixes for PERFECT-MODEL."""
+    from esmvalcore.cmor.fix import Fix
+
+    class tas(Fix):
+        """Fixes for tas variable.""""
+
+        def fix_metadata(self, cubes):
+            """
+            Fix metadata for tas.
+
+            Fix the name of the latitude coordinate, which is called altitude
+            in the original file.
+            """"
+            # Sometimes Iris will interpret the data as multiple cubes.
+            # Good CMOR datasets will only show one but we support the
+            # multiple cubes case to be able to fix the errors that are
+            # leading to that extra cubes.
+            # In our case this means that we can safely assume that the
+            # tas cube is the first one
+            tas_cube = cubes[0]
+            latitude = tas_cube.coord('altitude')
+
+            # Fix the names. Latitude values, units and
+            latitude.short_name = 'lat'
+            latitude.standard_name = 'latitude'
+            latitude.long_name = 'latitude'
+
+And that's all. The next time you run ESMValTool you will find that the error
```

Comment:
Sound a bit too informal, I would replace with: `This will fix the error and next time..."